File: backend/chat-service/src/main/java/com/example/chatservice/chat/repository/ChatRepository.java
package com.example.chatservice.chat.repository;

import com.example.chatservice.chat.entity.Chat;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface ChatRepository extends JpaRepository<Chat, Long>, ChatCustomRepository {

    List<Chat> findAllByIdIn(List<Long> ids);

    List<Chat> findAllByRoomId(String roomId);

    Page<Chat> findByRoomId(String roomId, Pageable pageable);

    Optional<Chat> findById(Long id);

    List<Chat> findAll();

    Long countByRoomId(String roomId);

    void deleteById(Long id);

    Long countByRoomIdAndIdBetween(String roomId, Long startId, Long endId);
}

File: backend/chat-service/src/main/kotlin/com/example/chatservice/chat/repository/ChatRepository.kt
package com.example.chatservice.chat.repository

import com.example.chatservice.chat.entity.Chat
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ChatRepository : JpaRepository<Chat, Long>, ChatCustomRepository {

    fun findAllByIdIn(ids: List<Long>): List<Chat>

    fun findAllByRoomId(roomId: String): List<Chat>

    fun findByRoomId(roomId: String, pageable: Pageable): Page<Chat>

    fun countByRoomId(roomId: String): Long

    fun countByRoomIdAndIdBetween(roomId: String, startId: Long, endId: Long): Long
}
